src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -344,9 +350,50 @@ async def __mount(self, pool, uuid, path=SYSDATASET_PATH):
                 os.mkdir(mountpoint)
             await run('mount', '-t', 'zfs', dataset, mountpoint, check=True)
 
+        if osc.IS_LINUX:
+
+            # make sure the glustereventsd webhook dir and
+            # config file exist and start the glustereventsd
+            # service (if appropriate)
```

Comment:
Just realized something... We will *always* start gluster and gluster eventsd?
Even when gluster has never been set up?

Reply:
I've changed the `eventsd.init` method to only setup the `/var/db/system/glusterd/events/webhooks.json` path if `glusterd` has been enabled.

Not sure how you want to handle `glusterd` and `ctdb_shared_vol` datasets that are setup in the sysdataset plugin always. If we don't set those up and user enables `glusterd` after system is booted, I'm assuming I'd have to run `sysdataset.setup` again?

Reply:
Creating dataset is fine, but it seems like we are doing more than that, including mounting the gluster filesystem?

We wouldnt expect to have any of that infrastructure related to gluster up and running if there is no gluster configured/enabled, especially because 90%+ of TrueNAS users won't use it

I may be misunderstanding what all these things do

Reply:
You're correct in that it was also creating a directory in the `glusterd` system dataset as well as a file and then writing certain data to that file and then starting the `glustereventsd` service. However, with my recent commit that doesn't occur unless `glusterd` is enabled in db like all other services.

Reply:
Thank you for clarifying.

Shouldn't we make a way to not even try to mount in this case then?

e.g. .init returns something that allows me to put whole block within an `if`?

Reply:
Good point